From 6c56a614ad7be9bd0d607173178ac4bcbe84f008 Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 13 Apr 2014 12:48:21 -0700
Subject: [PATCH] Format / update brainfuck plugin.

The only functional change in this is removing the usage of 'decode' on an str object, which breaks the plugin in python 3.
Besides that, it's all aesthetic changes

diff --git a/plugins/brainfuck.py b/plugins/brainfuck.py
index a7dc12e..3d82c8e 100644
--- a/plugins/brainfuck.py
+++ b/plugins/brainfuck.py
@@ -12,9 +12,11 @@ MAX_STEPS = 1000000
 
 
 @hook.command('brainfuck')
-@hook.command
+@hook.command()
 def bf(inp):
-    """bf <prog> -- Executes <prog> as Brainfuck code."""
+    """bf <prog> -- Executes <prog> as Brainfuck code.
+    :type inp: str
+    """
 
     program = re.sub('[^][<>+-.,]', '', inp)
 
@@ -30,17 +32,17 @@ def bf(inp):
                 brackets[open_brackets[-1]] = pos
                 open_brackets.pop()
             else:
-                return 'unbalanced brackets'
+                return "Unbalanced brackets"
     if len(open_brackets) != 0:
-        return 'unbalanced brackets'
+        return "Unbalanced brackets"
 
     # now we can start interpreting
-    ip = 0        # instruction pointer
-    mp = 0        # memory pointer
+    ip = 0  # instruction pointer
+    mp = 0  # memory pointer
     steps = 0
     memory = [0] * BUFFER_SIZE  # initial memory area
     rightmost = 0
-    output = ""   # we'll save the output here
+    output = ""  # we'll save the output here
 
     # the main program loop:
     while ip < len(program):
@@ -74,16 +76,16 @@ def bf(inp):
         ip += 1
         steps += 1
         if steps > MAX_STEPS:
-            if output == '':
-                output = '(no output)'
-            output += '[exceeded {} iterations]'.format(MAX_STEPS)
+            if not output:
+                output = "(no output)"
+            output += "(exceeded {} iterations)".format(MAX_STEPS)
             break
 
     stripped_output = re.sub(r'[\x00-\x1F]', '', output)
 
-    if stripped_output == '':
-        if output != '':
-            return 'no printable output'
-        return 'no output'
+    if not stripped_output:
+        if output:
+            return "No printable output"
+        return "No output"
 
-    return stripped_output[:430].decode('utf8', 'ignore')
+    return stripped_output[:430]
-- 
1.9.2

